src/Simulator/IncrementalRedstoneSimulator/RedstoneToggleHandler.h

```diff
@@ -77,14 +77,11 @@ class cRedstoneToggleHandler:
 
 	virtual unsigned char GetPowerLevel(cWorld & a_World, Vector3i a_Position, BLOCKTYPE a_BlockType, NIBBLETYPE a_Meta) const override
 	{
-		UNUSED(a_World);
-		UNUSED(a_Position);
-
 		switch (a_BlockType)
 		{
 			case E_BLOCK_LEVER: return cBlockLeverHandler::IsLeverOn(a_Meta) ? 15 : 0;
-			case E_BLOCK_STONE_BUTTON:
-			case E_BLOCK_WOODEN_BUTTON: return cBlockButtonHandler::IsButtonOn(a_Meta) ? 15 : 0;
+			case E_BLOCK_STONE_BUTTON: return cBlockButtonHandler::IsButtonOn(a_Meta) ? 15 : 0;
+			case E_BLOCK_WOODEN_BUTTON: return cBlockButtonHandler::IsButtonOn(a_World, a_Position, a_Meta) ? 15 : 0;
```

Comment:
This is very ugly, this code gives the impression that stone buttons can be on but not give out any redstone signal.

Reply:
To be honest, I don't see, why this code would say something like this, but I agree that there might be better ways to handle wodden and stone buttons differently. Just didn't want to give the blocktype to `IsButtonOn` too...

Reply:
I will push and you say, if it is better now - it is more like before...but only one function

Reply:
I can't see anything new having been pushed. The issue is the difference between IsButtonOn(meta) and IsButtonOn(meta, position). Why should changing position affect button on-ness. I think passing the BlockType in does make more sense.

Reply:
Ah I understand, okay :)
There should some stuff be pushed though :x